packages/dos-donts-block/src/components/ImageComponent.tsx

```diff
@@ -0,0 +1,154 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { Button, ButtonEmphasis, IconArrowCircleUp, IconImageStack, IconPlus32, LoadingCircle } from '@frontify/fondue';
+import { joinClassNames, radiusStyleMap, toRgbaString } from '@frontify/guideline-blocks-shared';
+import { useMemo } from 'react';
+import { DoDontImageHeight, IMAGE_HEIGHT_VALUES, ImageComponentProps, ImageFitChoice } from '../types';
+
+const ImageComponent = ({
```

Comment:
This component contains a lot of logic. Could you split it up and provide the properties in a react context, so you can avoid to pass down all the props?

Reply:
Hey, yeah I do realise there is some logic here, but I think it might still be acceptable in this case. The whole file is 100 lines, and it either shows the image or shows the button. How would you split it up? I'd be happy to do it if you have a nice suggestion.

Providing the props in react context would not reduce the lines too much, because I would still need to destructure it that would bloat the component either way, so I might prefer to keep it consistent to other components and keep the prop drilling.

Reply:
i get your point, let's keep it as props. as an improvement and to reduce the number of props, you could i.e define `border` as a prop instead [of having `borderColor`, `borderStyle` and `borderWidth`](https://github.com/Frontify/guideline-blocks/pull/445/files#diff-2de7de30c652f8231b1363e589b960fcf2387e09d4c49c316aeddd1f78a0bbf1R23-R25). wdyt?

Reply:
Yeah, good idea! Updated the PR.